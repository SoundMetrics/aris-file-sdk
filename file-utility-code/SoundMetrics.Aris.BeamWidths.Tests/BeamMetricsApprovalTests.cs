using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using SoundMetrics.Aris.Headers;
using System.Text;

namespace SoundMetrics.Aris.BeamWidths.Tests
{
    [TestClass]
    [UseApprovalSubdirectory("approval-files"), UseReporter(typeof(DiffReporter))]
    public sealed class BeamMetricsApprovalTests
    {
        [TestMethod]
        public void BeamWidths_ARIS3000_128()
            => RunTest(new BeamConfig(ArisSystemType.Aris3000, 128, LensType.None));

        [TestMethod]
        public void BeamWidths_ARIS3000_64()
            => RunTest(new BeamConfig(ArisSystemType.Aris3000, 64, LensType.None));

        [TestMethod]
        public void BeamWidths_ARIS1200_48()
            => RunTest(new BeamConfig(ArisSystemType.Aris1200, 48, LensType.None));

        [TestMethod]
        public void BeamWidths_ARIS1800_48()
            => RunTest(new BeamConfig(ArisSystemType.Aris1800, 48, LensType.None));

        [TestMethod]
        public void BeamWidths_ARIS1800_96()
            => RunTest(new BeamConfig(ArisSystemType.Aris1800, 96, LensType.None));

        [TestMethod]
        public void BeamWidths_ARIS1200_Telephoto_48()
            => RunTest(new BeamConfig(ArisSystemType.Aris1200, 48, LensType.Telephoto));

        [TestMethod]
        public void BeamWidths_ARIS1800_Telephoto_48()
            => RunTest(new BeamConfig(ArisSystemType.Aris1200, 48, LensType.Telephoto));

        [TestMethod]
        public void BeamWidths_ARIS1800_Telephoto_96()
            => RunTest(new BeamConfig(ArisSystemType.Aris1800, 96, LensType.Telephoto));

        private static void RunTest(BeamConfig cfg)
        {
            var stringOutput = BuildStringOutput(cfg);
            Approvals.Verify(stringOutput);
        }

        private static string BuildStringOutput(BeamConfig cfg)
        {
            var beamInfos = Metrics.GetBeamInformation(cfg);
            var stringOutput = BuildOutput(beamInfos);
            return stringOutput;

            static string BuildOutput(BeamInfo[] beamInfos)
            {
                var buffer = new StringBuilder();
                buffer.AppendLine($"# of BeamInfo members: [{beamInfos.Length}]");

                foreach (var beamInfo in beamInfos)
                {
                    buffer.AppendLine(beamInfo.ToString());
                }

                return buffer.ToString();
            }
        }
    }
}
