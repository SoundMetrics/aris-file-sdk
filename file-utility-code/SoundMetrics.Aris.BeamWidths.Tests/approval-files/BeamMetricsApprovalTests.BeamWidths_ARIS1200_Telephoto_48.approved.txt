# of BeamInfo members: [48]
BeamInfo { BeamNumber = 0, Center = -7.339, Left = -7.487, Right = -7.345 }
BeamInfo { BeamNumber = 1, Center = -7.043, Left = -7.193, Right = -7.048 }
BeamInfo { BeamNumber = 2, Center = -6.744, Left = -6.895, Right = -6.749 }
BeamInfo { BeamNumber = 3, Center = -6.443, Left = -6.595, Right = -6.447 }
BeamInfo { BeamNumber = 4, Center = -6.14, Left = -6.293, Right = -6.144 }
BeamInfo { BeamNumber = 5, Center = -5.835, Left = -5.988, Right = -5.839 }
BeamInfo { BeamNumber = 6, Center = -5.529, Left = -5.683, Right = -5.533 }
BeamInfo { BeamNumber = 7, Center = -5.222, Left = -5.377, Right = -5.225 }
BeamInfo { BeamNumber = 8, Center = -4.913, Left = -5.069, Right = -4.915 }
BeamInfo { BeamNumber = 9, Center = -4.602, Left = -4.758, Right = -4.604 }
BeamInfo { BeamNumber = 10, Center = -4.29, Left = -4.448, Right = -4.291 }
BeamInfo { BeamNumber = 11, Center = -3.975, Left = -4.133, Right = -3.976 }
BeamInfo { BeamNumber = 12, Center = -3.659, Left = -3.818, Right = -3.66 }
BeamInfo { BeamNumber = 13, Center = -3.342, Left = -3.501, Right = -3.343 }
BeamInfo { BeamNumber = 14, Center = -3.025, Left = -3.184, Right = -3.026 }
BeamInfo { BeamNumber = 15, Center = -2.708, Left = -2.867, Right = -2.709 }
BeamInfo { BeamNumber = 16, Center = -2.391, Left = -2.55, Right = -2.392 }
BeamInfo { BeamNumber = 17, Center = -2.074, Left = -2.233, Right = -2.075 }
BeamInfo { BeamNumber = 18, Center = -1.756, Left = -1.915, Right = -1.757 }
BeamInfo { BeamNumber = 19, Center = -1.438, Left = -1.598, Right = -1.438 }
BeamInfo { BeamNumber = 20, Center = -1.119, Left = -1.279, Right = -1.119 }
BeamInfo { BeamNumber = 21, Center = -0.8, Left = -0.96, Right = -0.8 }
BeamInfo { BeamNumber = 22, Center = -0.48, Left = -0.64, Right = -0.48 }
BeamInfo { BeamNumber = 23, Center = -0.16, Left = -0.32, Right = -0.16 }
BeamInfo { BeamNumber = 24, Center = 0.16, Left = 0.16, Right = 0.32 }
BeamInfo { BeamNumber = 25, Center = 0.48, Left = 0.48, Right = 0.64 }
BeamInfo { BeamNumber = 26, Center = 0.8, Left = 0.8, Right = 0.96 }
BeamInfo { BeamNumber = 27, Center = 1.119, Left = 1.119, Right = 1.279 }
BeamInfo { BeamNumber = 28, Center = 1.438, Left = 1.438, Right = 1.598 }
BeamInfo { BeamNumber = 29, Center = 1.756, Left = 1.757, Right = 1.915 }
BeamInfo { BeamNumber = 30, Center = 2.074, Left = 2.075, Right = 2.233 }
BeamInfo { BeamNumber = 31, Center = 2.391, Left = 2.392, Right = 2.55 }
BeamInfo { BeamNumber = 32, Center = 2.708, Left = 2.709, Right = 2.867 }
BeamInfo { BeamNumber = 33, Center = 3.025, Left = 3.026, Right = 3.184 }
BeamInfo { BeamNumber = 34, Center = 3.342, Left = 3.343, Right = 3.501 }
BeamInfo { BeamNumber = 35, Center = 3.659, Left = 3.66, Right = 3.818 }
BeamInfo { BeamNumber = 36, Center = 3.975, Left = 3.976, Right = 4.133 }
BeamInfo { BeamNumber = 37, Center = 4.29, Left = 4.291, Right = 4.448 }
BeamInfo { BeamNumber = 38, Center = 4.602, Left = 4.604, Right = 4.758 }
BeamInfo { BeamNumber = 39, Center = 4.913, Left = 4.915, Right = 5.069 }
BeamInfo { BeamNumber = 40, Center = 5.222, Left = 5.225, Right = 5.377 }
BeamInfo { BeamNumber = 41, Center = 5.529, Left = 5.533, Right = 5.683 }
BeamInfo { BeamNumber = 42, Center = 5.835, Left = 5.839, Right = 5.988 }
BeamInfo { BeamNumber = 43, Center = 6.14, Left = 6.144, Right = 6.293 }
BeamInfo { BeamNumber = 44, Center = 6.443, Left = 6.447, Right = 6.595 }
BeamInfo { BeamNumber = 45, Center = 6.744, Left = 6.749, Right = 6.895 }
BeamInfo { BeamNumber = 46, Center = 7.043, Left = 7.048, Right = 7.193 }
BeamInfo { BeamNumber = 47, Center = 7.339, Left = 7.345, Right = 7.487 }
